### PR TITLE
docs: grammar edits, sentence-case secondary headings

### PR DESCRIPTION
**Pull request by EricFromCanada**
A thorough review of the docs' wording and readability. Also changes some heading levels so only one primary heading appears per page. <sup>I hope no egos are bruised.</sup>

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [N/A] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [N/A] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: docs/Acceptable-Formulae.md [hunk 1]
```diff
@@ -78,9 +78,9 @@ etc.
 
 If not, then put bindings in the formula they bind to. This is more
 useful to people. Just install the stuff! Having to faff around with
-foo-ruby foo-perl etc. sucks.
+foo-ruby, foo-perl etc. sucks.
 
-### Niche (or self-submitted) Stuff<a name="Niche_Stuff"></a>
+### Niche (or self-submitted) stuff<a name="Niche_Stuff"></a>
```
Comment (reitermarkus) [inline docs/Acceptable-Formulae.md]: Remove the `<a name="Niche_Stuff"></a>` link here.

Comment (EricFromCanada) [inline docs/Acceptable-Formulae.md]: There's a few in FAQ.md as well, I guess they're no longer relevant?

Comment (MikeMcQuaid) [inline docs/Acceptable-Formulae.md]: Yeh. Would be great to figure out a way to have these be linkable on docs.brew.sh without manually specifying them, too.

Comment (EricFromCanada) [inline docs/Acceptable-Formulae.md]: Kramdown auto-generates IDs for each title, which can be linked to in lieu of `name` attributes, e.g. http://docs.brew.sh/Querying-Brew.html#examples

Comment (MikeMcQuaid) [inline docs/Acceptable-Formulae.md]: Thanks for that 🙇

**Comment by MikeMcQuaid** [comment]
Praise be to @EricFromCanada: King Of The Docs!

**Comment by EricFromCanada** [comment]
I'd like to thank Regular Expressions, without whom none of this would be possible...

**Comment by ilovezfs** [comment]
>Some people, when confronted with a problem, think “I know,
I'll use regular expressions.”  Now they have two problems.

http://regex.info/blog/2006-09-15/247

**Comment by EricFromCanada** [comment]
This programmer, when writing a reStructuredText highlighter, thought "I know, I'll use regular expressions." He had [many problems](https://bitbucket.org/EricFromCanada/ericfromcanada.bitbucket.org/src/tip/bbedit/reStructuredText.plist?fileviewer=file-view-default).

**Pull request merged**